src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXLegalityCheck.cpp

```diff
@@ -15,13 +15,36 @@
 
 #include "src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXLegalityCheck.hpp"
 #include "src/Accelerators/NNPA/Conversion/ONNXToZHigh/NNPALimit.h"
+#include "src/Compiler/CompilerOptions.hpp"
 #include "src/Conversion/ONNXToKrnl/RNN/RNNBase.hpp"
 #include "src/Dialect/ONNX/ONNXDimAnalysis.hpp"
 #include "src/Dialect/ONNX/ONNXOps/ShapeHelper.hpp"
 
 using namespace mlir;
 using namespace onnx_mlir;
 
+/// Convert the input NNPA level, ie. "z16", to a floating point value
+/// representing the level, ie. "16.0".
+float convertNNPALevel(std::string inputNNPALevel) {
+  float retNNPAFloat = 0;
+  try {
+    retNNPAFloat = std::strtof(
+        inputNNPALevel.substr(1, inputNNPALevel.size()).c_str(), NULL);
+  } catch (...) {
+    retNNPAFloat = 0;
+  }
+  return retNNPAFloat;
+}
+
+/// A function to check whether the input NNPA level, ie. "z16", is compatible
+/// with the current NNPA level.
+bool isCompatibleWithNNPALevel(std::string inputNNPALevel) {
+  if (convertNNPALevel(inputNNPALevel) <= convertNNPALevel(mcpu)) {
```

Comment:
For brevity, we can use `return convertNNPALevel(inputNNPALevel) <= convertNNPALevel(mcpu);`

BTW, a corner case, if both `convertNNPALevel(inputNNPALevel)` and `convertNNPALevel(cpu)` return 0 (because of some failure), this condition is `true` that is not what we want. Perhaps, you could check if `convertNNPALevel` succeeded or not before doing comparison.

Reply:
Thanks, I updated for the corner case and also brevity.